### ISSUE TITLE
Removing table header gets error with PDF output.

### ISSUE THREAD

**Issue by leovan** [haozhu233/kableExtra#196]
If I set the data.frame names to `NULL` or `col.names` in `knitr::kable` to `NULL`, it can remove the table header, and works pretty well in HTML. But it got errors with PDF output.

Briefly I found out **2 problems**:

1. Error in header_rows_start:header_rows_end : NA/NaN argument
Calls: <Anonymous> ... kable_styling -> pdfTable_styling -> styling_latex_striped
2. Error in add_indent_latex(out, seq(start_row, end_row)) :  There aren't that many rows in the table. Check positions in add_indent_latex.

1st problem is caused by `header_rows_end` in `styling_latex_striped()` in `kable_styling.R`, it gets NA with PDF output.
2nd problem is caused by `positions` in `add_indent_latex()` in `add_indent.R`, it seems when keeping the header, if the table gets `n` rows, `table_info$contents` has `n+1` elements with the header. If no header, `max(positions)` may equal to `check_nrows`, then it gets errors.

All the 2 problems are caused by rows counting I think. I try to get a quick fix, but after I fixed these 2 problems, it has some styling errors with PDF output which is caused by this fix. Sorry I still do not have time go over the entire project. Wish you can give a better solution.

Codes can reproduce the problems:

```{r}
library(tidyverse)
library(knitr)
library(kableExtra)

options(knitr.table.format = 'latex')

test_table <- tibble(
    col1 = c(1, 2, 3, 4)
)
names(test_table) <- c(NULL)
test_table %>%
    kable(booktabs = T, longtable = T) %>%
    kable_styling(latex_options = c('striped', 'hold_position', 'repeat_header')) %>%
    group_rows('TEST', 2, 4)
```

**Comment by haozhu233** [comment]
@leovan Yeah, I recognized this problem a while ago. As you said, it's sort of difficult to fix it in LaTeX. The main problem is that there are still a lot of people in favor of table without booktabs (or somehow they just don't use it). In those cases, there is no way for me to know (at the downstream of kable) whether there is a header row (since all rows end with hlines and they all look the same)....

Due to the scope of work and the complexity, I tends to leave this issue as is. I feel like the cases where tables don't have column header are sort of rare. If it doesn't work, adding a column header usually won't be a bad thing. What do you think? Of course, it would be really nice if you got time to fix it. ;)

**Comment by leovan** [comment]
Thanks for your reply, I usually use no-header table to do non-table things 😁, like listing. I just want to keep items in a table which can be referred and in a tight layout. I will go through this project later and try fix this.
